File: WePeiYang/lostfound/src/main/java/com/twtstudio/tjwhm/lostfound/mylist/MylistTableAdapter.java
package com.twtstudio.tjwhm.lostfound.mylist;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.twtstudio.tjwhm.lostfound.R;
import com.twtstudio.tjwhm.lostfound.R2;
import com.twtstudio.tjwhm.lostfound.detail.DetailActivity;
import com.twtstudio.tjwhm.lostfound.release.ReleaseActivity;
import com.twtstudio.tjwhm.lostfound.support.Utils;
import com.twtstudio.tjwhm.lostfound.waterfall.WaterfallBean;

import java.util.Objects;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by tjwhm on 2017/7/4.
 **/

public class MylistTableAdapter extends RecyclerView.Adapter {

    private MylistBean mylistBean;
    private Context context;
    private String lostOrFound;
    private MylistContract.MylistView mylistView;

    public MylistTableAdapter(MylistBean mylistBean, Context context, String lostOrFound, MylistContract.MylistView mylistView) {
        this.mylistBean = mylistBean;
        this.context = context;
        this.lostOrFound = lostOrFound;
        this.mylistView = mylistView;
    }

    public class MylistViewHolder extends RecyclerView.ViewHolder {
        @BindView(R2.id.mylist_item_status)
        TextView mylist_item_status;
        @BindView(R2.id.mylist_item_title)
        TextView mylist_item_title;
        @BindView(R2.id.mylist_item_type)
        TextView mylist_item_type;
        @BindView(R2.id.mylist_item_time)
        TextView mylist_item_time;
        @BindView(R2.id.mylist_item_place)
        TextView mylist_item_place;
        @BindView(R2.id.mylist_item_back_blue)
        ImageView mylist_item_back_blue;
        @BindView(R2.id.mylist_item_back_grey)
        ImageView mylist_item_back_grey;
        @BindView(R2.id.mylist_item_pencil)
        ImageView mylist_item_pencil;
        @BindView(R2.id.mylist_item_pic)
        ImageView mylist_item_pic;
        @BindView(R2.id.mylist_status_linear)
        LinearLayout mylist_status_linear;
        @BindView(R2.id.mylist_item_pencil_touch)
        TextView mylist_item_pencil_touch;

        public MylistViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view;
        view = LayoutInflater.from(context).inflate(R.layout.lf_item_mylist, parent, false);
        return new MylistViewHolder(view);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {

        MylistViewHolder viewHolder = (MylistViewHolder) holder;
        WaterfallBean.DataBean dataBean = mylistBean.data.get(position);
        viewHolder.mylist_item_title.setText(dataBean.title);
        viewHolder.mylist_item_type.setText(Utils.getType(dataBean.detail_type));
        viewHolder.mylist_item_time.setText(dataBean.time);
        viewHolder.mylist_item_place.setText(dataBean.place);
        Glide.with(context)
                .load(Utils.getPicUrl(dataBean.picture))
                .asBitmap()
                .placeholder(R.drawable.lf_waterfall_nopic)
                .into(viewHolder.mylist_item_pic);

        if (dataBean.isback == 1) {
            viewHolder.mylist_item_back_blue.setVisibility(View.VISIBLE);
            viewHolder.mylist_item_back_grey.setVisibility(View.GONE);
            if (Objects.equals(lostOrFound, "found")) {
                viewHolder.mylist_item_status.setText("已交还");
            } else {
                viewHolder.mylist_item_status.setText("已找到");
            }
        } else {
            viewHolder.mylist_item_back_blue.setVisibility(View.GONE);
            viewHolder.mylist_item_back_grey.setVisibility(View.VISIBLE);
            if (Objects.equals(lostOrFound, "found")) {
                viewHolder.mylist_item_status.setText("未交还");
            } else {
                viewHolder.mylist_item_status.setText("未找到");
            }
        }
        viewHolder.mylist_status_linear.setOnClickListener(view -> mylistView.turnStatus(dataBean.id));

        Intent intent = new Intent();
        Bundle bundle = new Bundle();
        viewHolder.itemView.setOnClickListener(view -> {
            bundle.putInt("id", dataBean.id);
            intent.putExtras(bundle);
            intent.setClass(context, DetailActivity.class);
            context.startActivity(intent);
        });
        viewHolder.mylist_item_pencil_touch.setOnClickListener(view -> {
            if (Objects.equals(lostOrFound, "lost")) {
                bundle.putString("lostOrFound", "editLost");
            } else {
                bundle.putString("lostOrFound", "editFound");
            }
            bundle.putInt("id", dataBean.id);
            bundle.putInt("type", dataBean.detail_type);
            intent.putExtras(bundle);
            intent.setClass(context, ReleaseActivity.class);
            context.startActivity(intent);
        });
    }

    @Override
    public int getItemCount() {
        return mylistBean.data.size();
    }
}

File: WePeiYang/lostfound2/src/main/java/com/example/lostfond2/mylist/MylistTableAdapter.kt
package com.example.lostfond2.mylist

import android.content.Intent
import android.os.Bundle
import android.support.v4.app.FragmentActivity
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.example.lostfond2.R
import com.example.lostfond2.detail.DetailActivity
import com.example.lostfond2.release.ReleaseActivity
import com.example.lostfond2.service.MyListDataOrSearchBean
import com.example.lostfond2.service.Utils
import com.twt.wepeiyang.commons.experimental.network.CommonBody


class MylistTableAdapter : RecyclerView.Adapter<MylistTableAdapter.MylistViewHolder> {


    var mylistBean: MutableList<MyListDataOrSearchBean>
    var context: FragmentActivity?
    var lostOrFound: String
    var mylistView: MyListService.MyListView

    constructor(mylistBean: MutableList<MyListDataOrSearchBean>, context: FragmentActivity?, lostOrFound: String, mylistView: MyListService.MyListView) {
        this.mylistBean = mylistBean
        this.context = context
        this.lostOrFound = lostOrFound
        this.mylistView = mylistView
    }


    class MylistViewHolder : RecyclerView.ViewHolder {
        constructor(view: View) : super(view) {
            mylist_item_status = view.findViewById(R.id.mylist_item_status)
            mylist_item_title = view.findViewById(R.id.mylist_item_title)
            mylist_item_type = view.findViewById(R.id.mylist_item_type)
            mylist_item_time = view.findViewById(R.id.mylist_item_time)
            mylist_item_place = view.findViewById(R.id.mylist_item_place)
            mylist_item_back_blue = view.findViewById(R.id.mylist_item_back_blue)
            mylist_item_back_grey = view.findViewById(R.id.mylist_item_back_grey)
            mylist_item_pencil = view.findViewById(R.id.mylist_item_pencil)
            mylist_item_pic = view.findViewById(R.id.mylist_item_pic)
            mylist_item_pencil_touch = view.findViewById(R.id.mylist_item_pencil_touch)


        }

        var mylist_item_status: TextView
        var mylist_item_title: TextView
        var mylist_item_type: TextView
        var mylist_item_time: TextView
        var mylist_item_place: TextView
        var mylist_item_back_blue: ImageView
        var mylist_item_back_grey: ImageView
        var mylist_item_pencil: ImageView
        var mylist_item_pic: ImageView

        var mylist_item_pencil_touch: TextView
    }

    override fun onBindViewHolder(holder: MylistViewHolder, position: Int) {
        val (id, _, title, place, time, _, detail_type, isback, picture) = mylistBean.get(position)
        holder.mylist_item_title.text = title
        holder.mylist_item_type.text = Utils.getType(detail_type)
        holder.mylist_item_time.text = time
        holder.mylist_item_place.text = place
        Glide.with(context)
                .load(Utils.getPicUrl(picture))
                .asBitmap()
                .placeholder(R.drawable.lf_waterfall_nopic)
                .into(holder.mylist_item_pic)

        if (isback == 1) {
            holder.mylist_item_back_blue.visibility = View.VISIBLE
            holder.mylist_item_back_grey.visibility = View.GONE
            if (lostOrFound == "found") {
                holder.mylist_item_status.text = "已交还"
            } else {
                holder.mylist_item_status.text = "已找到"
            }
        } else {
            holder.mylist_item_back_blue.visibility = View.GONE
            holder.mylist_item_back_grey.visibility = View.VISIBLE
            if (lostOrFound == "found") {
                holder.mylist_item_status.text = "未交还"
            } else {
                holder.mylist_item_status.text = "未找到"
            }
        }
        holder.mylist_item_status.setOnClickListener({ view -> mylistView.turnStatus(id) })

        val intent = Intent()
        val bundle = Bundle()
        holder.itemView.setOnClickListener { view ->
            bundle.putInt("id", id)
            intent.putExtras(bundle)
            intent.setClass(context, DetailActivity::class.java)
            context?.startActivity(intent)
        }
        holder.mylist_item_pencil_touch.setOnClickListener { view ->
            if (lostOrFound == "lost") {
                bundle.putString("lostOrFound", "editLost")
            } else {
                bundle.putString("lostOrFound", "editFound")
            }
            bundle.putInt("id", id)
            bundle.putInt("type", detail_type)
            intent.putExtras(bundle)
            intent.setClass(context, ReleaseActivity::class.java)
            context?.startActivity(intent)
        }

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MylistViewHolder {
        var view: View = LayoutInflater.from(context).inflate(R.layout.lf2_item_mylist, parent, false)
        return MylistViewHolder(view)
    }

    override fun getItemCount(): Int = mylistBean.size


}

